src/bundle/Core/Command/VirtualFieldDuplicateFixCommand.php

```diff
@@ -0,0 +1,272 @@
+
+        $this->addOption(
+            'sleep',
+            's',
+            InputOption::VALUE_OPTIONAL,
+            'Wait between iterations, in milliseconds',
+            self::DEFAULT_SLEEP
+        );
+
+        $this->addOption(
+            'force',
+            'f',
+            InputOption::VALUE_NONE,
+            'Force operation (implies non-interactive mode)',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $style = new SymfonyStyle($input, $output);
+        $stopwatch = new Stopwatch(true);
+        $stopwatch->start('total', 'command');
+
+        $force = $input->getOption('force');
+        if ($force) {
+            $input->setInteractive(false);
+        }
+
+        $batchSize = (int)$input->getOption('batch-size');
+        if ($batchSize === 0) {
+            $style->warning('Batch size is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $maxIterations = (int)$input->getOption('max-iterations');
+        if ($maxIterations === 0) {
+            $style->warning('Max iterations is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $sleep = (int)$input->getOption('sleep');
+
+        try {
+            $totalCount = $this->getDuplicatedAttributeTotalCount($style, $stopwatch);
+
+            if ($totalCount > 0) {
+                $confirmation = $this->askForConfirmation($style);
+                if (!$confirmation && !$force) {
+                    $style->info('Confirmation rejected. Terminating.');
+
+                    return Command::FAILURE;
+                }
+            } else {
+                $style->success('Database is clean of attribute duplicates. Nothing to do.');
+
+                return Command::SUCCESS;
+            }
+
+            $iteration = 1;
+            $totalDeleted = 0;
+            do {
+                $deleted = 0;
+                $stopwatch->start('iteration', 'sql');
+
+                $attributes = $this->getDuplicatedAttributesBatch($batchSize);
+                foreach ($attributes as $attribute) {
+                    $attributeIds = $this->getDuplicatedAttributeIds($attribute);
+
+                    $deleted += $this->deleteAttributes($attributeIds);
+                    $totalDeleted += $deleted;
+                }
+
+                $style->info(
+                    sprintf(
+                        'Iteration %d: Removed %d duplicates (total removed this execution: %d). [Debug %s]',
+                        $iteration,
+                        $deleted,
+                        $totalDeleted,
+                        $stopwatch->stop('iteration')
+                    )
+                );
+
+                if ($maxIterations !== self::MAX_ITERATIONS_UNLIMITED && ++$iteration > $maxIterations) {
+                    $style->warning('Max iterations count reached. Terminating.');
+
+                    return self::FAILURE;
```

Comment:
It's not a failure. It's expected execution state.
```suggestion
                    return self::SUCCESS;
```

Reply:
I have no strong feeling about this either way; it is matter of PoV. I intended this to be hard limit ("if there are more interations than max, interations exceed expectations", hence failure), not soft limit ("expect to do numer of iterations, without care how many more are there left", hence success). It is matter of intention, and my intention was to make it hard limit. Just how exceeding `max_execution_time` results in an error; I see this as similar case. I would say command should result in success only if there are no more duplicates left.

Reply:
Given that this command is never supposed to be called automatically (like in a bash command or something) I can agree to leave it as-is.

However, I disagree that you can assume that having duplicates remaining is a failure condition. The command did exactly what it was asked for.

In a real command we would probably discuss making this an input option, so it can be controlled, but I guess it does not matter in this context.